docs/current/sailbot_workspace/usage/help.md

```diff
@@ -0,0 +1,70 @@
+        - We recommend allocating at least 50 GB to Ubuntu to leave some wiggle room for Docker
+        - The process is similar for other Ubuntu and Windows versions,
+          but feel free to search for a guide specific to the combination you want to dual boot
+        - Since Sailbot Workspace uses Docker, it should be able to run on any Linux distribution, not just Ubuntu.
+          However, we may not be able to provide support if you encounter any difficulties with this
+
+## Troubleshooting
+
+If you are having some trouble running our software, here are some things you can try:
+
+### Sailbot Workspace Troubleshooting
+
+- [Update Sailbot Workspace](./workflow.md#2-update-sailbot-workspace){target=_blank}
+- Run the `setup` task to update package dependencies
+- Build from scratch
+    1. Run the `clean` task to delete C++ generated files
+    2. Run the `purge` task to delete ROS generated files
+    3. Run the `Build All` task to rebuild
+
+### VS Code Troubleshooting
+
+- Rebuild the Dev Container: run the `Dev Containers: Rebuild Container` VS Code command
+- Reload VS Code: run the `Developer: Reload Window` VS Code command
+- Identify broken extension: run the `Help: Start Extension Bisect` VS Code command
+    - Once you have identified a broken extension, you can [install a previous version](https://stackoverflow.com/a/53755378){target=_blank}
+      until the issue is fixed in a new release
+
+### System Troubleshooting
+
+- Restart WSL: close Sailbot Workspace and Docker Desktop then run `wsl --shutdown` in PowerShell
+- Restart computer
+
+### Docker Troubleshooting
+
+- Delete Docker files
+
+    ??? tip "Running Docker CLI commands on Windows"
+
+        On Windows, Docker CLI commands should be run in the Ubuntu terminal while Docker Desktop is running.
+
+    - Run `docker system prune` to remove all unused containers, networks, and dangling and unreferenced images
+        - Add `--all` to additionally remove unused images (don't have a container associated with them)
+        - Add `--volumes` to additionally remove volumes (makes Bash history and ROS logs persist across containers)
+    - Run `docker rmi -f $(docker images -aq)` to remove all images
+- [Install a previous version of Docker Desktop](https://stackoverflow.com/a/77224786){target=_blank}
+
+## Shrink WSL Distributions' Size
+
+After using Docker and Ubuntu for a while, you may notice that the vdisks are very large. As of May 2024,
+they are located at `C:\Users\<user>\AppData\Local\Docker\wsl\data\ext4.vhdx` and `C:\Users\<user>\AppData\Local\Packages\CanonicalGroupLimited.Ubuntu_79rhkp1fndgsc\LocalState\ext4.vhdx`,
+respectively.
+
+The problem is that these vdisks can automatically grow but not shrink, so if you download
+large files (like Docker images) and delete them once they're not needed the space is not freed.
+You can shrink vdisk using [these commands](https://github.com/microsoft/WSL/issues/4699#issuecomment-627133168){target=_blank}.
```

Comment:
That OptimizeVHD command isn't available by default (at least on Windows 11), I recall looking into it but hitting some friction to get access to it that I found the other alternatives like `wslcompact` easier (_although this had a gotcha of it's own IIRC_).

I've been using the WSL2 beta feature (_I'm not sure if it's since been stabilized_) to convert to sparse vdisks. These don't seem to always reclaim disk space (_at least from Docker_), but using tools like `docker system prune` with it does reclaim a fair amount back in my experience. Some users report the sparse feature does not work for them though.

Reply:
I documented the `diskpart` method, which has worked for me every time I have used it. Perhaps the corruption that one person encountered using this method was due to some conflict with reverting a sparse vdisk.

My thoughts on the other methods you mentioned:

- `wslcompact`: third party, requires installation, have not personally tried it
- sparse vdisks: still an experimental feature, have not personally tried it

Thus I am still recommending the `diskpart` method, especially to people who don't have a lot of experience with Windows/WSL